src/hotspot/share/opto/rangeinference.cpp

```diff
@@ -0,0 +1,1045 @@
+  } else if (t->is_con()) {
+    st->print("int:%s", intname(buf1, sizeof(buf1), t->get_con()));
+  } else if (int_type_is_equal(t, TypeInt::BOOL)) {
+    st->print("bool");
+  } else if (int_type_is_equal(t, TypeInt::BYTE)) {
+    st->print("byte");
+  } else if (int_type_is_equal(t, TypeInt::CHAR)) {
+    st->print("char");
+  } else if (int_type_is_equal(t, TypeInt::SHORT)) {
+    st->print("short");
+  } else {
+    if (verbose) {
+      st->print("int:%s..%s, %s..%s, %s",
+                intname(buf1, sizeof(buf1), t->_lo), intname(buf2, sizeof(buf2), t->_hi),
+                uintname(buf3, sizeof(buf3), t->_ulo), uintname(buf4, sizeof(buf4), t->_uhi),
+                bitname(buf5, sizeof(buf5), t->_bits._zeros, t->_bits._ones));
+    } else {
+      if (t->_lo >= 0) {
+        if (t->_hi == max_jint) {
+          st->print("int:>=%s", intname(buf1, sizeof(buf1), t->_lo));
+        } else {
+          st->print("int:%s..%s", intname(buf1, sizeof(buf1), t->_lo), intname(buf2, sizeof(buf2), t->_hi));
+        }
+      } else if (t->_hi < 0) {
+        if (t->_lo == min_jint) {
+          st->print("int:<=%s", intname(buf1, sizeof(buf1), t->_hi));
+        } else {
+          st->print("int:%s..%s", intname(buf1, sizeof(buf1), t->_lo), intname(buf2, sizeof(buf2), t->_hi));
+        }
+      } else {
+        st->print("int:%s..%s, %s..%s",
+                  intname(buf1, sizeof(buf1), t->_lo), intname(buf2, sizeof(buf2), t->_hi),
+                  uintname(buf3, sizeof(buf3), t->_ulo), uintname(buf4, sizeof(buf4), t->_uhi));
+      }
+
+    }
+  }
+
+  if (t->_widen > 0 && t != TypeInt::INT) {
+    st->print(", widen: %d", t->_widen);
+  }
+}
+
+void TypeIntHelper::int_type_dump(const TypeLong* t, outputStream* st, bool verbose) {
+  char buf1[80], buf2[80], buf3[80], buf4[80], buf5[80];
+  if (int_type_is_equal(t, TypeLong::LONG)) {
+    st->print("long");
+  } else if (t->is_con()) {
+    st->print("long:%s", longname(buf1, sizeof(buf1), t->get_con()));
+  } else {
+    if (verbose) {
+      st->print("long:%s..%s ^ %s..%s, bits:%s",
+                longname(buf1, sizeof(buf1), t->_lo), longname(buf2,sizeof(buf2), t-> _hi),
+                ulongname(buf3, sizeof(buf3), t->_ulo), ulongname(buf4, sizeof(buf4), t->_uhi),
+                bitname(buf5, sizeof(buf5), t->_bits._zeros, t->_bits._ones));
+    } else {
+      if (t->_lo >= 0) {
+        if (t->_hi == max_jint) {
+          st->print("long:>=%s", longname(buf1, sizeof(buf1), t->_lo));
+        } else {
+          st->print("long:%s..%s", longname(buf1, sizeof(buf1), t->_lo), longname(buf2, sizeof(buf2), t->_hi));
+        }
+      } else if (t->_hi < 0) {
+        if (t->_lo == min_jint) {
+          st->print("long:<=%s", longname(buf1, sizeof(buf1), t->_hi));
+        } else {
+          st->print("long:%s..%s", longname(buf1, sizeof(buf1), t->_lo), longname(buf2, sizeof(buf2), t->_hi));
+        }
+      } else {
+        st->print("long:%s..%s ^ %s..%s",
```

Comment:
```suggestion
        st->print("long:%s..%s, %s..%s",
```
That's what you have for the ints above.